src/typing_extensions.py

```diff
@@ -581,12 +609,12 @@ def _proto_hook(other):
                 if not cls.__dict__.get('_is_protocol', None):
                     return NotImplemented
                 if not getattr(cls, '_is_runtime_protocol', False):
-                    if _caller(depth=3) in {'abc', 'functools'}:
+                    if _allow_reckless_class_checks():
```

Comment:
Is it correct that the depth is changed from 3 to 4?

Reply:
Yes, because by adding a new function that the code has to pass through before it gets to the `sys._getframe` call, the call stack becomes "another frame deep"

Reply:
I could just have looked at `_caller` and figured that out myself. 🤦